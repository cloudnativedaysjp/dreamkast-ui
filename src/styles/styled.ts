import styled from 'styled-components'
import { Grid } from '@material-ui/core'

export const Container = styled.div`
  height: 100%;
  background: rgba(255, 255, 255, 0.7);
  font-size: 1.2em;
  border-radius: 10px;
  margin: 5px;
  padding: 10px;
`

export const OuterContainer = styled(Grid)`
  height: 100%;
`

export const CenterizedContainer = styled.div`
  text-align: center;
`

export const Header2 = styled.h2<{ centerized?: boolean }>`
  color: #037f8c;
  ${({ centerized }) => (centerized ? 'text-align: center' : '')};
`

export const Header3 = styled.h3<{ centerized?: boolean }>`
  color: #037f8c;
  ${({ centerized }) => (centerized ? 'text-align: center' : '')};
`

export const MiroIframe = styled.iframe`
  width: 100%;
  min-height: 500px;
`

export const MenuLink = styled.a`
  text-decoration-line: none;
`

export const ButtonContainer = styled.div`
  padding-bottom: 10px;
`
