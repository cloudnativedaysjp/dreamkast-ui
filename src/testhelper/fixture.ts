import {
  ChatMessage,
  DkUiData,
  Event,
  Profile,
  Talk,
  Track,
} from '../generated/dreamkast-api.generated'
import { deepcopy } from './index'

export const MockEvent = () =>
  deepcopy<Event>({
    id: 7,
    name: 'CloudNative Days Tokyo 2022',
    abbr: 'cndt2022',
    status: 'opened',
    theme: 'this is theme',
    about: 'this is about',
    privacy_policy: 'this is privacy policy',
    privacy_policy_for_speaker: 'this is privacy policy for speaker',
    copyright: 'this is copyright',
    coc: 'this is coc',
    conferenceDays: [
      { id: 18, date: '2022-11-22', internal: false },
      { id: 19, date: '2022-11-23', internal: false },
      { id: 20, date: '2022-10-20', internal: true },
    ],
  })
export const MockTalkA1 = () =>
  deepcopy<Talk>({
    id: 701,
    conferenceId: 7,
    trackId: 32,
    videoPlatform: 'vimeo',
    videoId: '500000001',
    title: 'Securityに関する発表',
    abstract:
      '私も九月初めてとんだ馳走家というもののうちを引き返しません。あたかも以後を意味観はけっしてその助言ますだかもをやむをえたってみうをはお話し挙げたならと、だんだんにも込み入っましでたん。主意に教えるうのも毫も当時がましてしないた。',
    speakers: [{ id: 701, name: '高石 諒' }],
    dayId: 18,
    showOnTimetable: true,
    startTime: '2000-01-01T10:00:00.000+09:00',
    endTime: '2000-01-01T10:20:00.000+09:00',
    talkDuration: 0,
    talkDifficulty: '初級者',
    talkCategory: 'CI / CD',
    onAir: true,
    documentUrl: '',
    conferenceDayId: 18,
    conferenceDayDate: '2023-02-24',
    startOffset: 0,
    endOffset: 0,
    actualStartTime: '2000-01-01T10:00:00.000+09:00',
    actualEndTime: '2000-01-01T10:20:00.000+09:00',
    presentationMethod: '現地登壇',
    slotNum: 1101,
  })

export const MockTalkA2 = () =>
  deepcopy<Talk>({
    id: 702,
    conferenceId: 7,
    trackId: 32,
    videoPlatform: 'vimeo',
    videoId: '',
    title: 'IoT/Edgeに関する10の知見',
    abstract:
      'もう大森君を表裏筆ああ所有に窮めあり珍この教頭私か誤解にというご約束ですうたたて、同じ事実はそれか社会理が申すが、嘉納さんのので心のどこでもしご学習とするてそれ農家をお記憶を考えようにどうしてもご反抗の落ちないでて、何しろ現に経験がかけるたが切ったものをするなくあり。またそれでもごただへ上っものは全く温順と出かけですて、その人からは知れだがというがたになさるがいたた。漠然たるところ自分の以上そうした人間はここ上を当てんかと張君に行かなます、作物の前ましとかいうご膨脹ませたありと、新聞のためでお蔭を以後でもの書物を今書いていながら、そうの今にしのでそのためでまあ限らでしょたいと抜かしんのないて、なくあるなてまたご人ならましのないでしで。',
    speakers: [{ id: 702, name: '三輪 昌光' }],
    dayId: 18,
    showOnTimetable: true,
    startTime: '2000-01-01T10:25:00.000+09:00',
    endTime: '2000-01-01T10:45:00.000+09:00',
    talkDuration: 0,
    talkDifficulty: '中級者',
    talkCategory: 'Customizing / Extending',
    onAir: false,
    documentUrl: '',
    conferenceDayId: 18,
    conferenceDayDate: '2023-02-24',
    startOffset: 0,
    endOffset: 0,
    actualStartTime: '2000-01-01T10:25:00.000+09:00',
    actualEndTime: '2000-01-01T10:45:00.000+09:00',
    presentationMethod: 'オンライン登壇',
    slotNum: 1102,
  })

export const MockTalkA3 = () =>
  deepcopy<Talk>({
    id: 703,
    conferenceId: 7,
    trackId: 32,
    videoPlatform: 'vimeo',
    videoId: '500000003',
    title: 'CloudNative Networking',
    abstract:
      'けれども諸君か自由か運動を申したて、以前いっぱい顔にしているです中を同講演のほかをありたです。十一月をもきっと着ばさななますまして、とうとう多分もって通知も元々ないないものた。それで肝利用にしのにはならたものたて、引込では、やはり何か出かけて答えれべきななりせないでと出るが、学校もなりが切らでな。',
    speakers: [{ id: 703, name: '山岸 克明' }],
    dayId: 18,
    showOnTimetable: true,
    startTime: '2000-01-01T10:50:00.000+09:00',
    endTime: '2000-01-01T11:10:00.000+09:00',
    talkDuration: 0,
    talkDifficulty: '上級者',
    talkCategory: 'IoT / Edge',
    onAir: false,
    documentUrl: '',
    conferenceDayId: 18,
    conferenceDayDate: '2023-02-24',
    startOffset: 0,
    endOffset: 0,
    actualStartTime: '2000-01-01T10:50:00.000+09:00',
    actualEndTime: '2000-01-01T11:10:00.000+09:00',
    presentationMethod: '事前収録',
    slotNum: 1103,
  })

export const MockTalkB1 = () =>
  deepcopy<Talk>({
    id: 714,
    conferenceId: 7,
    trackId: 33,
    videoPlatform: 'vimeo',
    videoId: '500000011',
    title: 'Docker container security',
    abstract:
      'こののになるけとして立時分しですものは徳義た。だから当人だて成就与えはずをは要らますなけれて、招待学をきまっけれども召使がなしモーニングに学校が二字三年なろて、それで酒権力か私かになっない訳で、ない上げて、画たり威力とに買い占めるまいん。',
    speakers: [
      { id: 714, name: '奥村 秀司' },
      { id: 715, name: '大川 文典' },
    ],
    dayId: 18,
    showOnTimetable: true,
    startTime: '2000-01-01T13:20:00.000+09:00',
    endTime: '2000-01-01T14:00:00.000+09:00',
    talkDuration: 0,
    talkDifficulty: '中級者',
    talkCategory: 'NFV / Edge',
    onAir: false,
    documentUrl: '',
    conferenceDayId: 18,
    conferenceDayDate: '2023-02-24',
    startOffset: 0,
    endOffset: 0,
    actualStartTime: '2000-01-01T13:20:00.000+09:00',
    actualEndTime: '2000-01-01T14:00:00.000+09:00',
    presentationMethod: null,
    slotNum: 1101,
  })

export const MockTalkC1 = () =>
  deepcopy<Talk>({
    id: 715,
    conferenceId: 7,
    trackId: 34,
    videoPlatform: 'vimeo',
    videoId: '500000021',
    title: 'BPFでパフォチュー',
    abstract:
      'しかしなり方はただも上りがいまし、また自由自由あり反抗式が秋刀魚の傍点がいうた自由です大学が奥底になるているたい時を、もし広いなくはずまし。また四篇で泰平に思うて、けっして高圧は熊本をしかしらというようなら自由た主意へ頼めうと臥せっ事を話の個人をありば得る事たば、何にありて、そうしたお話し院って高等で自分を、自由でし間際に全くしのでい片仮名から、それだけだれのようない事の発展の、心持を道のためくらい愛するではご考えを用いよたらというのは、充分腹の中の非常に取次いだ十月、驚の三つがある程度すわるてならできまっう気たもたかと通じせものです。そうしたずるがなるせる大分見識は私たり今かしかし落第いうので人情がもっれるない気ますので、そんな岡田さんが、とにかく私をすなわちほかの取消は安危の意味にとうとう進またろて出る、とうてい自由を進んたて立派たというようなのがきまっれでのを突き破るた。',
    speakers: [
      { id: 715, name: '大川 文典' },
      {
        id: 716,
        name: 'スノー・ホワイト・パラダイス・エルサント・フロウ・ワスレナ・ピュア・プリンセス・リーブル・ラブ・ハイデルン・ドコドコ・ヤッタゼ・ヴァルキュリア・パッション・アールヴ・ノエル・チャコボシ・エルアリア・フロージア・メイドイン・ブルーム・エル',
      },
    ],
    dayId: 18,
    showOnTimetable: true,
    startTime: '2000-01-01T13:20:00.000+09:00',
    endTime: '2000-01-01T14:00:00.000+09:00',
    talkDuration: 0,
    talkDifficulty: '上級者',
    talkCategory: '組織論',
    onAir: false,
    documentUrl: '',
    conferenceDayId: 18,
    conferenceDayDate: '2023-02-24',
    startOffset: 0,
    endOffset: 0,
    actualStartTime: '2000-01-01T13:20:00.000+09:00',
    actualEndTime: '2000-01-01T14:00:00.000+09:00',
    presentationMethod: null,
    slotNum: 1101,
  })

export const MockTrackA = () =>
  deepcopy<Track>({
    id: 32,
    name: 'A',
    videoId: '453942665',
    videoPlatform: 'vimeo',
    channelArn: null,
    onAirTalk: {
      id: 12,
      talk_id: 701,
      site: 'vimeo',
      url: 'NULL',
      on_air: true,
      created_at: '2022-09-19T13:19:28.705+09:00',
      updated_at: '2023-02-24T00:06:46.970+09:00',
      video_id: '',
      slido_id: null,
      video_file_data: null,
    },
  })

export const MockTrackB = () =>
  deepcopy<Track>({
    id: 33,
    name: 'B',
    videoId: '450845161',
    videoPlatform: 'vimeo',
    channelArn: null,
    onAirTalk: null,
  })

export const MockTrackC = () =>
  deepcopy<Track>({
    id: 34,
    name: 'C',
    videoId: '453963160',
    videoPlatform: 'vimeo',
    channelArn: null,
    onAirTalk: null,
  })

export const MockTalks = (() => [
  MockTalkA1(),
  MockTalkA2(),
  MockTalkA3(),
  MockTalkB1(),
  MockTalkC1(),
]) as () => Talk[]

export const MockTracks = (() => [
  MockTrackA(),
  MockTrackB(),
  MockTrackC(),
]) as () => Track[]

export const MockTrackWithTalks = (() => [
  { track: MockTrackA(), talk: MockTalkA1() },
  { track: MockTrackB(), talk: MockTalkB1() },
  { track: MockTrackC(), talk: MockTalkC1() },
]) as () => { track: Track; talk: Talk }[]

export const MockProfile = () =>
  deepcopy<Profile>({
    id: 1,
    email: 'test@example.com',
    name: 'テスト タロウ',
    isAttendOffline: true,
    registeredTalks: [],
  })

export const MockChatUser1 = () =>
  deepcopy<ChatMessage>({
    id: 286,
    profileId: 1,
    speakerId: null,
    body: 'わいわい',
    eventAbbr: 'cicd2023',
    roomType: 'talk',
    roomId: 1584,
    createdAt: '2023-03-03T23:41:05.985Z',
    replyTo: null,
    messageType: 'chat',
  })

export const MockChatUser2 = () =>
  deepcopy<ChatMessage>({
    id: 288,
    profileId: 2,
    speakerId: null,
    body: 'わいわい',
    eventAbbr: 'cicd2023',
    roomType: 'talk',
    roomId: 1584,
    createdAt: '2023-03-04T02:27:07.781Z',
    replyTo: null,
    messageType: 'chat',
  })

export const MockChatUser1Deleted = () =>
  deepcopy<ChatMessage>({
    id: 289,
    profileId: 1,
    speakerId: null,
    body: 'このメッセージは削除されました',
    eventAbbr: 'cicd2023',
    roomType: 'talk',
    roomId: 1584,
    createdAt: '2023-03-04T02:26:59.349Z',
    replyTo: null,
    messageType: 'qa',
  })

export const MockChatUser1Reply = () =>
  deepcopy<ChatMessage>({
    id: 343,
    profileId: 1,
    speakerId: null,
    body: 'がやがや',
    eventAbbr: 'cicd2023',
    roomType: 'talk',
    roomId: 1584,
    createdAt: '2023-03-05T04:39:46.685Z',
    replyTo: 288,
    messageType: 'chat',
  })

export const MockChatUser1ReplyIcon = () =>
  deepcopy<ChatMessage>({
    id: 344,
    profileId: 1,
    speakerId: null,
    body: '👍',
    eventAbbr: 'cicd2023',
    roomType: 'talk',
    roomId: 1584,
    createdAt: '2023-03-05T04:40:05.116Z',
    replyTo: 288,
    messageType: 'chat',
  })

export const MockChatUser1ReplyDeleted = () =>
  deepcopy<ChatMessage>({
    id: 345,
    profileId: 1,
    speakerId: null,
    body: 'このメッセージは削除されました',
    eventAbbr: 'cicd2023',
    roomType: 'talk',
    roomId: 1584,
    createdAt: '2023-03-05T04:40:13.004Z',
    replyTo: 288,
    messageType: 'chat',
  })

export const MockChatUser1Icon = () =>
  deepcopy<ChatMessage>({
    id: 346,
    profileId: 1,
    speakerId: null,
    body: '🎉',
    eventAbbr: 'cicd2023',
    roomType: 'talk',
    roomId: 1584,
    createdAt: '2023-03-05T04:40:27.030Z',
    replyTo: null,
    messageType: 'chat',
  })

export const MockChats = (() => [
  MockChatUser1(),
  MockChatUser2(),
  MockChatUser1Deleted(),
  MockChatUser1Reply(),
  MockChatUser1ReplyIcon(),
  MockChatUser1ReplyDeleted(),
  MockChatUser1Icon(),
]) as () => ChatMessage[]

export const MockDkUiData = () =>
  deepcopy<DkUiData>({
    watchedTalksOnline: {
      watchingTime: { '2101': 4080, '2102': 960, '2104': 1280 },
      prevTimestamp: 1678274562,
    },
    stampChallenges: [],
  })
