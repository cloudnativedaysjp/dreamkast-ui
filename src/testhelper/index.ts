export function deepcopy<T>(src: T): T {
  return JSON.parse(JSON.stringify(src))
}
