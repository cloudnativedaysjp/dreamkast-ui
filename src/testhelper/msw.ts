import type { RequestHandler } from 'msw'
import { setupServer } from 'msw/node'

export function setupMockServer(...handlers: RequestHandler[]) {
  const server = setupServer(...handlers)
  beforeAll(() => server.listen())
  afterEach(() => server.resetHandlers())
  afterAll(() => server.close())
  return server
}
