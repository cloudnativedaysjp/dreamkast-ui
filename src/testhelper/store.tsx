// ref: https://redux.js.org/usage/writing-tests

import React, { PropsWithChildren } from 'react'
import { combineReducers, PreloadedState } from 'redux'
import { configureStore } from '@reduxjs/toolkit'
import { dreamkastApi } from '../generated/dreamkast-api.generated'
import auth from '../store/auth'
import settings from '../store/settings'
import appData from '../store/appData'
import { setupListeners } from '@reduxjs/toolkit/query'
import { render, RenderOptions } from '@testing-library/react'
import { Provider } from 'react-redux'
import { RootState } from '../store'
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client'

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(dreamkastApi.middleware),
    preloadedState,
  })
  setupListeners(store.dispatch)
  return store
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {
      auth: auth.getInitialState(),
      settings: settings.getInitialState(),
      appData: appData.getInitialState(),
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      api: {} as any,
    },
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions,
) {
  function Wrapper({ children }: PropsWithChildren): JSX.Element {
    return <Provider store={store}>{children}</Provider>
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export function renderWithApolloClient(
  ui: React.ReactElement,
  renderOptions?: RenderOptions,
) {
  function Wrapper({ children }: PropsWithChildren): JSX.Element {
    const client = new ApolloClient({
      uri: 'http://localhost:3001',
      cache: new InMemoryCache(),
    })
    return <ApolloProvider client={client}>{children}</ApolloProvider>
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

const rootReducer = combineReducers({
  [dreamkastApi.reducerPath]: dreamkastApi.reducer,
  [auth.name]: auth.reducer,
  [settings.name]: settings.reducer,
  [appData.name]: appData.reducer,
})

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: ReturnType<typeof setupStore>
}
