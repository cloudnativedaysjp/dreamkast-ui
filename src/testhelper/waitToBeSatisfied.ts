export const waitToBeSatisfied = async (
  fn: () => boolean,
  timeout = 1000,
  interval = 10,
): Promise<boolean> => {
  return await new Promise((resolve, reject) => {
    let spent = 0
    setInterval(() => {
      if (fn()) {
        resolve(true)
        return
      }
      spent += interval
      if (spent > timeout) {
        reject()
        return
      }
    }, interval)
  })
}
