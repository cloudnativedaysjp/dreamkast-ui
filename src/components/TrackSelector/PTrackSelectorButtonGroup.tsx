import React from 'react'
import * as Styled from './styled'
import { Talk, Track } from '../../generated/dreamkast-api.generated'
import { Theme, Tooltip } from '@material-ui/core'
import dayjs from 'dayjs'
import { withStyles } from '@material-ui/styles'

type Props = {
  data: { track: Track; talk?: Talk }[]
  selectedTrack: number
  onChange: (selectedItem: number | null) => void
}

export const PTrackSelectorButtonGroup: React.FC<Props> = ({
  data,
  selectedTrack,
  onChange,
}) => {
  return (
    <Styled.TrackSelectorButtonGroup
      value={selectedTrack}
      color="primary"
      onChange={(_, i) => onChange(i)}
      exclusive
    >
      {data.map(({ track, talk }) => (
        <HtmlTooltip
          key={track.id}
          title={
            <React.Fragment>
              <div>
                {!talk ? (
                  'ライブセッションはありません'
                ) : (
                  <>
                    {talk.onAir && <Styled.Live>LIVE</Styled.Live>}{' '}
                    {dayjs(talk.startTime).format('HH:mm')}-
                    {dayjs(talk.endTime).format('HH:mm')}
                    <br />
                    {talk.title}
                    <br />
                    {talk.speakers?.map((s) => s.name).join(', ')}
                  </>
                )}
              </div>
            </React.Fragment>
          }
        >
          <Styled.MenuItem key={track.id} value={track.id}>
            {track.name}
          </Styled.MenuItem>
        </HtmlTooltip>
      ))}
    </Styled.TrackSelectorButtonGroup>
  )
}

const HtmlTooltip = withStyles((theme: Theme) => ({
  tooltip: {
    backgroundColor: '#f5f5f9',
    color: 'rgba(0, 0, 0, 0.87)',
    maxWidth: 220,
    fontSize: theme.typography.pxToRem(12),
    border: '1px solid #dadde9',
  },
}))(Tooltip)
