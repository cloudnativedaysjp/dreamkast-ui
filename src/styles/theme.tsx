import { createMuiTheme } from '@material-ui/core/styles'
import red from '@material-ui/core/colors/red'

// Create a theme instance.
const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#423A57',
    },
    secondary: {
      main: '#9A7FB8',
    },
    error: {
      main: red.A400,
    },
    background: {
      default: '#fff',
    },
  },
})

export default theme
