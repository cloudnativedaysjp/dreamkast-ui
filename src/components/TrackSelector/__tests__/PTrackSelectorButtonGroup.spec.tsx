import React from 'react'
import { fireEvent, render } from '@testing-library/react'

import { MockTrackWithTalks } from '../../../testhelper/fixture'
import { PTrackSelectorButtonGroup } from '../PTrackSelectorButtonGroup'
import theme from '../../../styles/theme'
import { ThemeProvider as MUIThemeProvider } from '@material-ui/styles'

describe('PTrackSelectorButtonGroup', function () {
  it('should render without crash', () => {
    const data = MockTrackWithTalks()
    render(
      <MUIThemeProvider theme={theme}>
        <PTrackSelectorButtonGroup
          data={data}
          selectedTrack={0}
          onChange={() => null}
        />
      </MUIThemeProvider>,
    )
  })

  it('should render when data empty', () => {
    render(
      <MUIThemeProvider theme={theme}>
        <PTrackSelectorButtonGroup
          data={[]}
          selectedTrack={0}
          onChange={() => null}
        />
      </MUIThemeProvider>,
    )
  })

  it('should call onChange callback when track selected', () => {
    const data = MockTrackWithTalks()
    const spy = jest.fn()
    const screen = render(
      <MUIThemeProvider theme={theme}>
        <PTrackSelectorButtonGroup
          data={data}
          selectedTrack={0}
          onChange={spy}
        />
      </MUIThemeProvider>,
    )
    fireEvent.click(screen.getByText(data[0].track.name))

    expect(spy).toHaveBeenCalledWith(data[0].track.id)
  })
})
