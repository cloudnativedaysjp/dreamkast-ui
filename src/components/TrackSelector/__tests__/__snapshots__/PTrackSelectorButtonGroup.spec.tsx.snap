// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PTrackSelectorButtonGroup should render when data empty 1`] = `
<DocumentFragment>
  <div
    class="MuiToggleButtonGroup-root styled__TrackSelectorButtonGroup-sc-hs19f1-1 bqZiMD"
    color="primary"
    role="group"
  />
</DocumentFragment>
`;

exports[`PTrackSelectorButtonGroup should render without crash 1`] = `
<DocumentFragment>
  <div
    class="MuiToggleButtonGroup-root styled__TrackSelectorButtonGroup-sc-hs19f1-1 bqZiMD"
    color="primary"
    role="group"
  >
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root styled__MenuItem-sc-hs19f1-2 faSDkg MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal"
      tabindex="0"
      type="button"
      value="32"
    >
      <span
        class="MuiToggleButton-label"
      >
        A
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root styled__MenuItem-sc-hs19f1-2 faSDkg MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal"
      tabindex="0"
      type="button"
      value="33"
    >
      <span
        class="MuiToggleButton-label"
      >
        B
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root styled__MenuItem-sc-hs19f1-2 faSDkg MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal"
      tabindex="0"
      type="button"
      value="34"
    >
      <span
        class="MuiToggleButton-label"
      >
        C
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
